.github/workflows/post_release_version_bump.yml

```diff
@@ -0,0 +1,52 @@
+name: Post-Release Version Bump
+
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  post_release_version_bump:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Derive versions
+        id: versions
+        run: |
+          old_version=`grep -oP '(?<=__version__ = ").*(?=")' xir/_version.py`
+          new_version=`echo $old_version | awk '{split($0,v,"."); print v[1] "." v[2]+1 "." v[3] "-dev"}'`
+
+          echo "::set-output name=old_version::$old_version"
+          echo "::set-output name=new_version::$new_version"
+
+      - name: Bump version in _version.py
+        run:
+          sed -i 's/${{ steps.versions.outputs.old_version }}/${{ steps.versions.outputs.new_version }}/' xir/_version.py
+
+      - name: Bump version in changelog
+        run: |
+          next_release_version=${{ steps.versions.outputs.new_version }}
+          next_release_version=${next_release_version%-dev}
+
+          header="## Release ${next_release_version} (development release)
+
+          ### Contributors
+
+          This release contains contributions from (in alphabetical order):
+          "
+
+          printf '%s\n%s\n' "$header" "$(cat .github/CHANGELOG.md)" > .github/CHANGELOG.md
```

Comment:
Oh, so this line takes two newlines + the header + the old changelog and replaces the changelog with it (basically adding the header to the top). Why the `printf` and the two `\n` at the top?

Reply:
Hmm, not quite! It replaces the changelog with the concatenation of:
1. `$header`
2. `\n`
3. the old changelog
4. `\n`

I've updated the command to hopefully make things a little more clear.

Reply:
Ooh, yes, of course! That's what the `%s` are for. 😄